packages/labextension/src/widget.tsx

```diff
@@ -0,0 +1,245 @@
+
+  useInterval(async () => {
+    if (kernelId && panel.isVisible) {
+      requestUsage(kernelId)
+        .then((usage) => setUsage(usage))
+        .catch(() => {
+          console.warn(`Request failed for ${kernelId}. Kernel restarting?`);
+        });
+    }
+  }, POLL_INTERVAL_SEC * 1000);
+
+  const requestUsage = (kid: string) => {
+    return requestAPI<any>(`get_usage/${kid}`).then((data) => {
+      const usage: Usage = {
+        ...data.content,
+        kernelId: kid,
+        timestamp: new Date(),
+      };
+      return usage;
+    });
+  };
+
+  useEffect(() => {
+    const createKernelChangeCallback = (panel: NotebookPanel) => {
+      return (
+        _sender: ISessionContext,
+        args: IChangedArgs<
+          Kernel.IKernelConnection | null,
+          Kernel.IKernelConnection | null,
+          'kernel'
+        >
+      ) => {
+        const newKernelId = args.newValue?.id;
+        if (newKernelId) {
+          setKernelId(newKernelId);
+          const path = panel?.sessionContext.session?.model.path;
+          setPath(path);
+          requestUsage(newKernelId).then((usage) => setUsage(usage));
+        } else {
+          // Kernel was disposed
+          setKernelId(newKernelId);
+        }
+      };
+    };
+
+    const notebookChangeCallback = (
+      sender: INotebookTracker,
+      panel: NotebookPanel | null
+    ) => {
+      if (panel === null) {
+        // Ideally we would switch to a new "select a notebook to get kernel
+        // usage" screen instead of showing outdated info.
+        return;
+      }
+      if (kernelChangeCallback) {
+        kernelChangeCallback.panel.sessionContext.kernelChanged.disconnect(
+          kernelChangeCallback.callback
+        );
+      }
+      kernelChangeCallback = {
+        callback: createKernelChangeCallback(panel),
+        panel,
+      };
+      panel.sessionContext.kernelChanged.connect(kernelChangeCallback.callback);
+
+      if (panel.sessionContext.session?.kernel?.id !== kernelId) {
+        const kernelId = panel.sessionContext.session?.kernel?.id;
+        if (kernelId) {
+          setKernelId(kernelId);
+          const path = panel.sessionContext.session?.model.path;
+          setPath(path);
+          requestUsage(kernelId).then((usage) => setUsage(usage));
+        }
+      }
+    };
+    props.currentNotebookChanged.connect(notebookChangeCallback);
+    return () => {
+      props.currentNotebookChanged.disconnect(notebookChangeCallback);
+      // In the ideal world we would disconnect kernelChangeCallback from
+      // last panel here, but this can lead to a race condition. Instead,
+      // we make sure there is ever only one callback active by holding
+      // it in a global state.
+    };
+  }, [kernelId]);
+
+  if (kernelId) {
+    if (usage) {
+      return !usage.hostname ? (
+        <>
+          <h3 className="jp-KernelUsage-section-separator">
+            Kernel usage details are not available
```

Comment:
It would be nice to also translate the string in the kernel usage panel.

Reply:
https://github.com/jupyter-server/jupyter-resource-usage/pull/164/commits/c65103dbcaed03f77730b72912e5d928904a5aeb translates `panel.ts`.
There is some text to translate in `widget.tsx`, but since it's in React elements, I'm not sure if it's possible?

Reply:
React elements can be translated the same way as any other code, you can pass the trans bundle in props and use `this.props.trans` or `props.trans` (in addition to standard `trans`, `this.trans` and `this._trans`). Just replace string literals like `<div>test</div>` with subsitutions like `<div>{props.trans.__('test')}</div>`